### PR TITLE
Use unique column names in aggregate-intersection

### PR DESCRIPTION
**Pull request by jgoizueta**
See #330

### CODE REVIEW

**Comment by jgoizueta** [comment]
I'm worried now that maybe using the id from params as a columns name suffix wasn't a good idea: if it is not used to compute the node hash id (is it?) then we could have different queries (differing in result column names) with the same identifier...

File: lib/node/column-namer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                ok = this.isUnique(name);
+            }
+        }
+        this.columns.push(name); // ?
+        return name;
+    }
+
+    isUnique(name) {
+        // TODO: should this be case-insensitive?
+        return !this.columns.includes(name);
```
Comment (jgoizueta) [inline lib/node/column-namer.js]: Can we use  `includes`?

**Comment by jgoizueta** [comment]
Hey @dgaubert, I've added a test to check aggregate-intersection does not produce duplicate column names. It has payed off, because I had missed another potential duplicate columns (`count_vals_density`). I've also checked the test fails because of duplicate `count_vals` with the original aggregate-intersection.

Please take a look when you can.

**Comment by jgoizueta** [comment]
TODO: we should check all analyses that inherit columns from other nodes and use the ColumnNamer for any columns they add.

File: lib/node/column-namer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+module.exports = class ColumnNamer {
+    constructor (columns) {
+        this.columns = columns.map(name => stripTableName(name).toLowerCase());
+    }
+
+    uniqueName(baseName) {
+        baseName = baseName.toLowerCase();
+        var name = baseName;
+        var ok = this.isUnique(name);
+        if (!ok) {
```
Comment (dgaubert) [inline lib/node/column-namer.js]: we don't need this initial checking, do we?. I mean if `while(this.isUnique)` is false it skips immediately of 
 suffixing column name, right?

File: lib/node/nodes/aggregate-intersection.js [hunk 3]
```diff
@@ -12,34 +12,46 @@ var PARAMS = {
...
+    var columnNamer = new ColumnNamer(columns);
+
     if (this.aggregate_column === null) {
         qualifiedAggregateColumn += '*';
     } else {
         qualifiedAggregateColumn += this.aggregate_column;
         finalColumnName += '_' + this.aggregate_column;
     }
 
+    if (this.aggregate_function === 'count') {
```
Comment (dgaubert) [inline lib/node/nodes/aggregate-intersection.js]: Can we extract everything related to assign a unique name to the output column to a its own method?

File: lib/node/column-namer.js (hunk 2 above)
Comment (jgoizueta) [reply lib/node/column-namer.js]: Right! (It was needed before I remove the code to use params.id and didn't realize it wasn't needed anymore)

File: lib/node/nodes/aggregate-intersection.js (hunk 3 above)
Comment (jgoizueta) [reply lib/node/nodes/aggregate-intersection.js]: That line isn't really related to the column renaming, it copes with the different name schema used for `count` aggregations. I moved that bit from where it was to be able to adjust the column name after it was computed.

So we already have all the column-ranaming stuff in this single line:
```
finalColumnName = columnNamer.uniqueName(finalColumnName);
```
